Firestore/core/src/firebase/firestore/model/field_value.cc

```diff
@@ -438,6 +500,45 @@ ComparisonResult FieldValue::CompareTo(const FieldValue& rhs) const {
   }
 }
 
+std::string FieldValue::ToString() const {
+  switch (tag_) {
+    case Type::Null:
+      return util::ToString(nullptr);
+    case Type::Boolean:
+      return util::ToString(boolean_value_);
+    case Type::Integer:
+      return util::ToString(integer_value_);
+    case Type::Double:
+      return util::ToString(double_value_);
+    case Type::Timestamp:
+      return util::ToString(*timestamp_value_);
+    case Type::ServerTimestamp:
+      return util::ToString(*server_timestamp_value_);
+    case Type::String:
+      return util::ToString(*string_value_);
+    case Type::Blob:
+      return absl::StrCat(
+          "<", absl::BytesToHexString(blob_value_as_string_view()), ">");
+    case Type::Reference:
+      return util::ToString(*reference_value_);
+    case Type::GeoPoint:
+      return util::ToString(*geo_point_value_);
+    case Type::Array:
+      return util::ToString(*array_value_);
+    case Type::Object:
+      return util::ToString(*object_value_);
+    default:
+      HARD_FAIL("Unsupported type %s", type());
```

Comment:
Question: `HARD_FAIL` is used instead of `UNREACHABLE` because of the more informative error message, right?